samples/MqttClient_Hello/app/application.cpp

```diff
@@ -74,23 +74,40 @@ void onMessageReceived(String topic, String message)
 void startMqttClient()
 {
 	procTimer.stop();
-	if(!mqtt->setWill("last/will","The connection from this device is lost:(", 1, true)) {
+	if(!mqtt.setWill("last/will","The connection from this device is lost:(", 1, true)) {
 		debugf("Unable to set the last will and testament. Most probably there is not enough memory on the device.");
 	}
-	mqtt->connect("esp8266", MQTT_USERNAME, MQTT_PWD, true);
+	
+	mqtt.setCallback(onMessageReceived);
+	URL url;
+	url.Protocol = "mqtt";
+#ifdef ENABLE_SSL
```

Comment:
Hm... If SSL is enabled that does not mean that you want to use SSL connection. Additionally the port is different for SSL and non-SSL connections.